app/uk/gov/hmrc/gform/formtemplate/RepeatingComponentService.scala

```diff
@@ -81,23 +81,42 @@ object RepeatingComponentService {
     }
 
     section.copy(
-      title = buildText(Some(section.title), index, section.fieldToTrack.get, data).getOrElse(""),
-      shortName = buildText(section.shortName, index, section.fieldToTrack.get, data),
+      title = buildText(Some(section.title), index, data).getOrElse(""),
+      shortName = buildText(section.shortName, index, data),
       fields = section.fields.map(copyField)
     )
   }
 
-  private def buildText(template: Option[String], index: Int, fieldToTrack: VariableInContext,
-    data: Map[String, String]): Option[String] = {
+  private def buildText(template: Option[String], index: Int, data: Map[String, String]): Option[String] = {
+
+    def evaluateTextExpression(str: String) = {
```

Comment:
This code looks very magic for me. Is there a way to make it more understandable?